Sources/PointFree/Episode/Show.swift

```diff
@@ -13,47 +13,48 @@ import Tuple
 
 let episodeResponse =
   filterMap(
-    over1(episode(forParam:)) >>> require1 >>> pure,
+    over1(episodes(current:)) >>> require1 >>> pure,
     or: writeStatus(.notFound) >=> respond(episodeNotFoundView.contramap(lower))
     )
     <| writeStatus(.ok)
     >=> userEpisodePermission
     >=> map(lower)
     >>> respond(
       view: episodeView,
-      layoutData: { permission, episode, currentUser, subscriberState, currentRoute in
+      layoutData: { permission, episodes, currentUser, subscriberState, currentRoute in
         let navStyle: NavStyle = currentUser == nil ? .mountains(.main) : .minimal(.light)
 
         return SimplePageLayoutData(
           currentRoute: currentRoute,
           currentSubscriberState: subscriberState,
           currentUser: currentUser,
-          data: (permission, currentUser, subscriberState, episode),
-          description: episode.blurb,
+          data: (permission, currentUser, subscriberState, episodes),
+          description: episodes.current.blurb,
           extraStyles: markdownBlockStyles <> pricingExtraStyles,
-          image: episode.image,
+          image: episodes.current.image,
           style: .base(navStyle),
-          title: "Episode #\(episode.sequence): \(episode.title)",
+          title: "Episode #\(episodes.current.sequence): \(episodes.current.title)",
           usePrismJs: true
         )
     }
 )
 
 let useCreditResponse =
   filterMap(
-    over1(episode(forParam:)) >>> require1 >>> pure,
+    over1(episodes(current:)) >>> require1 >>> pure,
     or: writeStatus(.notFound) >=> respond(episodeNotFoundView.contramap(lower))
     )
     <<< { userEpisodePermission >=> $0 }
+//    <<< map(map(over2(^\.current)))
```

Comment:
Tried getting this to work to avoid having to make `validateCreditRequest` care about `Zipper<Episode>` but was fighting the type system a lot.

Reply:
I think this is the first time we've done this exact thing, though a version of it is kinda baked into `filterMap` already.

we could accomplish this by defining `contramap` on middlewares:

```swift
((A) -> B) -> (Middleware<I, J, B, C>) -> Middleware<I, J, A, C>
```

and then we could do `<<< contramap(over2(^\.current))`.